specs/_features/maxeb_increase/capella.py

```diff
@@ -1776,6 +1788,49 @@ def process_execution_layer_withdraw_request(
     ))
 
 
+def is_consolidated(validator: Validator) -> bool:
+   validator.consolidated_to != UNSET_CONSOLIDATED_TO
+
+
+def resolve_slashed_index(state: BeaconState, index: ValidatorIndex) -> ValidatorIndex:
+    if is_consolidated(state.validators[index]):
+        # Recursively resolve consolidated index
+        return resolve_slashed_index(state, state.validators[index].consolidated_to)
+    else:
+        return index
+
+
+def process_consolidation(state: BeaconState, consolidation: Consolidation) -> None:
+    target_validator = state[consolidation.target_index]
+    source_validator = state[consolidation.source_index]
+
+    assert target_validator.exit_epoch == FAR_FUTURE_EPOCH
+    assert source_validator.exit_epoch == FAR_FUTURE_EPOCH
+
+    # verify source withdrawal credentials, which have authority over validating keys
+    assert source_validator.withdrawal_credentials[:1] == ETH1_ADDRESS_WITHDRAWAL_PREFIX
+    assert source_validator.withdrawal_credentials[12:] == consolidation.source_address
+
+    # target validator accepts consolidation operation onto itself
+    # Fork-agnostic domain since address changes are valid across forks
+    domain = compute_domain(DOMAIN_CONSOLIDATION, genesis_validators_root=state.genesis_validators_root)
+    signing_root = compute_signing_root(source_validator.pubkey, domain)
+    assert bls.Verify(target_validator.pubkey, signing_root, consolidation.target_signature)
+
+    active_balance = min(state.balances[consolidation.source_index], get_max_effective_balance(source_validator))
+    excess_balance = state.balances[consolidation.source_index] - active_balance
+    state.balances[consolidation.target_index] += active_balance
+    state.balances[consolidation.source_index] = 0
+
+    # Excess balance above current active balance ceil, send to inbound churn
+    state.pending_balance_deposits.append(PendingBalanceDeposit(consolidation.target_index, excess_balance))
+
+    source_validator.consolidated_to = consolidation.target_index
+    # Balance is not exiting the active set, do not apply churn
+    source_validator.exit_epoch = get_current_epoch(state)
```

Comment:
Note: instant exit breaks a `MAX_SEED_LOOKAHEAD` invariant. I don’t see anything is immediately broken by that but might worth keeping in mind. Target validator can also be made liable to the source’s duties if that need be (if the protocol had any duties assigned to a validator for the `MAX_SEED_LOOKAHEAD` epochs into the future).